File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/entities/Notice.java
package mx.edu.upqroo.kristenandroid.data.database.entities;

import com.google.gson.annotations.SerializedName;

import java.util.Date;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "notice")
public class Notice {

    @PrimaryKey()
    @SerializedName("idAvisos")
    @ColumnInfo(name = "noticeId")
    @NonNull
    private String noticeId;

    @SerializedName("titulo")
    @ColumnInfo(name = "title")
    private String title;

    @SerializedName("cuerpo")
    @ColumnInfo(name = "body")
    private String body;

    @SerializedName("fecha")
    @ColumnInfo(name = "date")
    private Date date;

    @SerializedName("idUsuarios")
    @ColumnInfo(name = "userId")
    private String userId;

    @SerializedName("idCarrera")
    @ColumnInfo(name = "careerId")
    private int careerId;

    public Notice(@NonNull String noticeId,
                  String title,
                  String body,
                  Date date,
                  String userId,
                  int careerId) {
        this.noticeId = noticeId;
        this.title = title;
        this.body = body;
        this.date = date;
        this.userId = userId;
        this.careerId = careerId;
    }

    @NonNull
    public String getNoticeId() {
        return noticeId;
    }

    public void setNoticeId(@NonNull String noticeId) {
        this.noticeId = noticeId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public int getCareerId() {
        return careerId;
    }

    public void setCareerId(int careerId) {
        this.careerId = careerId;
    }
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/entities/Notice.kt
package mx.edu.upqroo.kristenandroid.data.database.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import java.util.*

@Entity(tableName = "notice")
class Notice(@field:PrimaryKey
             @field:SerializedName("idAvisos")
             @field:ColumnInfo(name = "noticeId")
                var noticeId: String,
             @field:SerializedName("titulo")
             @field:ColumnInfo(name = "title")
                var title: String?,
             @field:SerializedName("cuerpo")
             @field:ColumnInfo(name = "body")
                var body: String?,
             @field:SerializedName("fecha")
             @field:ColumnInfo(name = "date")
                var date: Date?,
             @field:SerializedName("idUsuarios")
             @field:ColumnInfo(name = "userId")
                var userId: String?,
             @field:SerializedName("idCarrera")
             @field:ColumnInfo(name = "careerId")
                var careerId: Int)
